10.2/docker-entrypoint.sh

```diff
@@ -258,24 +258,31 @@ docker_setup_db() {
 	fi
 	# Sets root password and creates root users for non-localhost hosts
 	local rootCreate=
+	# SQL escape the root password, \ followed by '
+	local rootPasswordEscaped=${MARIADB_ROOT_PASSWORD//\\/\\\\}
+	rootPasswordEscaped=${rootPasswordEscaped//\'/\\\'}
+
 	# default root to listen for connections from anywhere
 	if [ -n "$MARIADB_ROOT_HOST" ] && [ "$MARIADB_ROOT_HOST" != 'localhost' ]; then
 		# no, we don't care if read finds a terminating character in this heredoc
 		# https://unix.stackexchange.com/questions/265149/why-is-set-o-errexit-breaking-this-read-heredoc-expression/265151#265151
 		read -r -d '' rootCreate <<-EOSQL || true
-			CREATE USER 'root'@'${MARIADB_ROOT_HOST}' IDENTIFIED BY '${MARIADB_ROOT_PASSWORD}' ;
+			CREATE USER 'root'@'${MARIADB_ROOT_HOST}' IDENTIFIED BY '${rootPasswordEscaped}' ;
```

Comment:
Does this one also need the `NO_BACKSLASH_ESCAPES` you added below?

Reply:
its in the same session where `rootCreate` is used.